test/metabase/models/setting_test.clj

```diff
@@ -513,3 +513,31 @@
     ;; ok, make sure the setting was set
     (is (= "Banana Beak"
            (toucan-name)))))
+
+
+(deftest duplicated-setting-name
+  (testing "can re-register a setting in the same ns (redefining or reloading ns)"
+    (is (defsetting foo (deferred-tru "A testing setting") :visibility :public))
+    (is (defsetting foo (deferred-tru "A testing setting") :visibility :public)))
+  (testing "if attempt to register in a different ns throws an error"
+    (let [current-ns (ns-name *ns*)]
+      (try
+        (ns nested-setting-test
+          (:require [metabase.models.setting :refer [defsetting]]
+                    [metabase.util.i18n :as i18n :refer [deferred-tru]]))
```

Comment:
Can't you just bind `*ns*` here? It seems a little funky putting an `ns` declaration inside a test

Reply:
The test needs another namespace, and the most reliable way to do that is to create and be in the namespace. I'm not sure if i can just bind `*ns*`. I know changing namespaces changes that binding but I honestly don't know if binding that dynamic var is sufficient to "be in" another ns. I suppose i could do `(in-ns 'nested-setting-test)` but then i'd need to require everything i need (including clojure) and that's what the `ns` form does anyways.